src/Microsoft.AspNetCore.Rewrite/UrlRewrite/FileParser.cs

```diff
@@ -0,0 +1,236 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml.Linq;
+
+namespace Microsoft.AspNetCore.Rewrite.UrlRewrite
+{
+
+    public class FileParser
```

Comment:
Name it `XmlFileParser` or something more descriptive because `FileParser` means it can parse any file